modules/search-service-impl/src/main/java/org/opencastproject/search/impl/SearchServiceImpl.java

```diff
@@ -734,4 +758,41 @@ public void updated(@SuppressWarnings("rawtypes") Dictionary properties) throws
     addJobLoad = LoadUtil.getConfiguredLoadValue(properties, ADD_JOB_LOAD_KEY, DEFAULT_ADD_JOB_LOAD, serviceRegistry);
     deleteJobLoad = LoadUtil.getConfiguredLoadValue(properties, DELETE_JOB_LOAD_KEY, DEFAULT_DELETE_JOB_LOAD, serviceRegistry);
   }
+
+  @Override
+  public List<Pattern> getProtectedUrlPattern() {
+    return Collections.singletonList(staticFilePattern);
+  }
+
+  private boolean loadUrlAccess(final String mediaPackageId) {
+    logger.debug("Check if user `{}` has access to media package `{}`", securityService.getUser(), mediaPackageId);
+    final SearchQuery query = new SearchQuery()
+        .withId(mediaPackageId)
+        .includeEpisodes(true)
+        .includeSeries(false);
+    return getByQuery(query).size() > 0;
+  }
+
+  @Override
+  public boolean verifyUrlAccess(final String path) {
+    // Always allow access for admin
+    final User user = securityService.getUser();
+    if (user.hasRole(GLOBAL_ADMIN_ROLE)) {
+      logger.debug("Allow access for admin `{}`", user);
+      return true;
+    }
+
+    final Matcher m = staticFilePattern.matcher(path);
+    if (!m.matches()) {
+      return false;
+    }
```

Comment:
A DEBUG line in the failed pattern match and in the failed org match might save some sites some headaches.

Reply:
Please refrain from complaining about any minor things at the moment. This code is meant to show how an implementation of this proposal would look like. There is more missing and complaining about minor stuff right now will serve no purpose except for distracting from the main question of this idea being acceptable or not ;-P

Reply:
Obviously, if you find a problem in the method's architecture, please complain!